File: src/java/tusan/nodes/expression/Term.java
package net.tutla.tums.tusan.nodes.expression;

import net.tutla.tums.tusan.Node;
import net.tutla.tums.tusan.lexer.Token;
import net.tutla.tums.tusan.lexer.TokenType;

import java.util.Arrays;
import java.util.Objects;

public class Term extends Node {
    public Object value;
    public Term(Token token){
        super(token);
    }

    public Term create(){
        Factor term1 = new Factor(token).create();
        if (interpreter.getNextToken().type == TokenType.OPERATOR){
            Token op = interpreter.getNextToken();
            if (Arrays.asList("*","/","**","^").contains(op.value)){
                interpreter.nextToken();
                Expression term2 = new Expression(interpreter.nextToken()).create();

                if (Objects.equals(op.value, "*")){
                    if (term1.value instanceof String && term2.value instanceof Double){
                        value = ((String) term1.value).repeat((int) term2.value);
                    } else {
                        assert term1.value instanceof Double;
                        value = (Double) term1.value * (Double) term2.value;
                    }
                } else if (Objects.equals(op.value, "/")) {
                    value = (Double) term1.value / (Double) term2.value;
                } else if (Objects.equals(op.value, "^")) {
                    value = (double) ((int) term1.value ^ (int) term2.value);
                } else if (Objects.equals(op.value, "**")) {
                    value = Math.pow((Double) term1.value, (Double) term2.value);
                }
            } else {
                value = term1.value;
            }
        } else {
            value = term1.value;
        }
        return this;
    }
}

File: src/python/src/tusan/nodes/expression/Term.py
from tusan.Node import *

class TermNode(Node):
    def __init__(self, factor:Token,rules=[]):
        self.auto_eval = True
        self.interpreter = factor.interpreter
        self.rules = rules
        self.returned_var = None
        
        self.factor = factor

    async def create(self):
        tkn1 = await FactorNode(self.factor,rules=self.rules).create()
        
        if self.interpreter.get_next_token().type == "OPERATOR":
            operator = self.interpreter.get_next_token()
            if operator.value in ["*","/","**","^"]:
                operator = self.interpreter.next_token()
                tkn2 = await ExpressionNode(self.interpreter.next_token()).create()
                if operator.value == "*":
                    self.value = tkn1.value * tkn2.value
                elif operator.value == "/":
                    self.value = tkn1.value / tkn2.value
                elif operator.value == "^":
                    self.value = int(tkn1.value)^int(tkn2.value)
            else:
                self.value = tkn1.value
        else:
            self.value = tkn1.value
        return self